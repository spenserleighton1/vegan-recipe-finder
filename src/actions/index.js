export const addRecipes = (recipes) => ({
  type: 'ADD_RECIPES',
  recipes
})

export const searchByIngredient = (ingredients) => ({
  type: 'SEARCH_BY_INGREDIENT', 
  ingredients
})