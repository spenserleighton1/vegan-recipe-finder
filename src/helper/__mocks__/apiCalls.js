export const fetchRecipes = jest.fn().mockImplementation(() => ({
  recipes: [{title: 'a food'}, {title: 'more a food'}, {title: 'even more a food'}]
}));