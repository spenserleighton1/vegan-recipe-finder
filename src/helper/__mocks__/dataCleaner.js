export const cleanData = jest.fn();

export const cleanRecipe = jest.fn();