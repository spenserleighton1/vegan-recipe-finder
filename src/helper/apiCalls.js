export const fetchRecipes = async (key) => {
  const url = `https://food2fork.com/api/search?key=${key}&q=vegan`;
  const response = await fetch(url);
  const results = await response.json();
  return results;
}