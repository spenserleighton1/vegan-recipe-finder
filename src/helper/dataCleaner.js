export const cleanData = data => {
  const cleanData = data.map(recipe => {
    return {     
      recipeTitle: recipe.title,
      recipeLink: recipe.source_url,
      recipeImage: recipe.image_url,
      recipeId: recipe.recipe_id }
    })
  return cleanData;
}

export const cleanRecipe = data => ({
  title: data.recipe.title,
  publisher: data.recipe.publisher,
  ingredients: data.recipe.ingredients,
  linkUrl: data.recipe.source_url,
  id: data.recipe.recipe_id 
})
