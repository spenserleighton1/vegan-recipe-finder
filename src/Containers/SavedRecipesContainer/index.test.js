import React from 'react';
import { shallow } from 'enzyme';
import { SavedRecipesContainer, mapStateToProps, mapDispatchToProps } from './';
import { fetchSingleRecipe } from '../../helper/apiCalls';
import { cleanRecipe} from '../../helper/dataCleaner';
import { addSavedRecipes, saveRecipe, deleteRecipe } from '../../actions';

jest.mock('../../helper/apiCalls');
jest.mock('../../helper/dataCleaner');

describe('SavedRecipesContainer', () => {
  let wrapper;
  let mockRecipes;
  let mockSaved;
  let mockAddSaved;
  let mockAuthUser;
  let mockSaveRecipe

  beforeEach(() => {
    mockAuthUser = { uid: 666 };
    mockSaveRecipe = jest.fn();
    mockAddSaved = jest.fn();
    mockSaved = ['123', '456'];
    mockRecipes = [{title: 'tired'},{title: 'bedtime'}];
    wrapper = shallow(<SavedRecipesContainer
      authUser={ mockAuthUser }
      saveRecipe={ mockSaveRecipe }
      addSavedRecipes={ mockAddSaved }
      savedRecipesFull={ mockRecipes } 
      savedRecipeIDs={ mockSaved }/>)
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('matches the snapshot when there are no superSavedRecipes', () => {
    const wrapper = shallow(<SavedRecipesContainer
      authUser={ mockAuthUser }
      saveRecipe={ mockSaveRecipe }
      addSavedRecipes={ mockAddSaved }
      savedRecipesFull={ [] } 
      savedRecipeIDs={ mockSaved }/>)

    expect(wrapper).toMatchSnapshot()
  })

  it('should invoke addSavedRecipes when fetchRecipe is called', async () => {
    const results = await wrapper.instance().fetchRecipes();

    expect(mockAddSaved).toHaveBeenCalled()
  })

  it('should call saveRecipe when addFavs is called', () => {
    wrapper.instance().addFavorite()

    expect(mockSaveRecipe).toHaveBeenCalled()
  })
})

  describe('mapDispatchToProps', () => {
    it('should return a props object with addSavedRecipes', () => {
      const mockDispatch = jest.fn();

      const mappedProps = mapDispatchToProps(mockDispatch)
      const actionToDispatch = addSavedRecipes({ title: 'a recipe' });

      mappedProps.addSavedRecipes({ title: 'a recipe' });
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('should return a props object with saveRecipe', () => {
      const mockDispatch = jest.fn();

      const mappedProps = mapDispatchToProps(mockDispatch)
      const actionToDispatch = saveRecipe({ id: 666});

      mappedProps.saveRecipe({ id: 666});
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('should return a props object with deleteRecipe', () => {
      const mockDispatch = jest.fn();

      const mappedProps = mapDispatchToProps(mockDispatch)
      const actionToDispatch = deleteRecipe({ id: 666});

      mappedProps.deleteRecipe({ id: 666});
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })

  describe('mapStateToProps', () => {
    it('should return a props object', () => {
      const mockState = {"savedRecipeIDs": [], "savedRecipesFull": []};
      const expected = {"savedRecipeIDs": [], "savedRecipesFull": []}

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  })