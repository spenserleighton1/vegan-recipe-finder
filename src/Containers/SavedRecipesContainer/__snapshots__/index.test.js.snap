// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedRecipesContainer matches the snapshot when there are no superSavedRecipes 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SavedRecipesContainer
    addSavedRecipes={[Function]}
    authUser={
        Object {
            "uid": 666,
          }
    }
    saveRecipe={[Function]}
    savedRecipeIDs={
        Array [
            "123",
            "456",
          ]
    }
    savedRecipesFull={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SavedRecipesContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SavedRecipesContainer
    addSavedRecipes={[Function]}
    authUser={
        Object {
            "uid": 666,
          }
    }
    saveRecipe={[Function]}
    savedRecipeIDs={
        Array [
            "123",
            "456",
          ]
    }
    savedRecipesFull={
        Array [
            Object {
              "title": "tired",
            },
            Object {
              "title": "bedtime",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          className="home-btn"
          exact={true}
          replace={false}
          to="/"
>
          Home
</Link>,
        Array [
          <SavedRecipeCard
            authUser={
                        Object {
                                    "uid": 666,
                                  }
            }
            title="tired"
/>,
          <SavedRecipeCard
            authUser={
                        Object {
                                    "uid": 666,
                                  }
            }
            title="bedtime"
/>,
        ],
      ],
      "className": "saved-recipes-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Home",
          "className": "home-btn",
          "exact": true,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": "Home",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "authUser": Object {
            "uid": 666,
          },
          "title": "tired",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "authUser": Object {
            "uid": 666,
          },
          "title": "bedtime",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="home-btn"
            exact={true}
            replace={false}
            to="/"
>
            Home
</Link>,
          Array [
            <SavedRecipeCard
              authUser={
                            Object {
                                          "uid": 666,
                                        }
              }
              title="tired"
/>,
            <SavedRecipeCard
              authUser={
                            Object {
                                          "uid": 666,
                                        }
              }
              title="bedtime"
/>,
          ],
        ],
        "className": "saved-recipes-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Home",
            "className": "home-btn",
            "exact": true,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Home",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "authUser": Object {
              "uid": 666,
            },
            "title": "tired",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "authUser": Object {
              "uid": 666,
            },
            "title": "bedtime",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
